File: app/src/main/java/com/mirash/passkeeper/db/CredentialsRepository.java
package com.mirash.passkeeper.db;

import androidx.lifecycle.LiveData;

import java.util.List;

/**
 * @author Mirash
 */
public class CredentialsRepository {
    private final CredentialsDao dao;
    private final CredentialsDatabase database;

    public CredentialsRepository(CredentialsDatabase database) {
        this.database = database;
        this.dao = database.getCredentialsDao();
    }

    public void insertAll(List<Credentials> credentialsList) {
        dao.insertAll(credentialsList);
    }

    public void updateAll(List<Credentials> credentialsList) {
        dao.updateAll(credentialsList);
    }

    public void insert(Credentials Credentials) {
        dao.insert(Credentials);
    }

    public void update(Credentials Credentials) {
        dao.update(Credentials);
    }

    public LiveData<List<Credentials>> getAllCredentialsLiveData() {
        return dao.getAll();
    }

    public LiveData<Credentials> getCredentialsById(int id) {
        return dao.getById(id);
    }

    public Credentials getCredentialsByIdSync(int id) {
        return dao.getByIdSync(id);
    }

    public List<Credentials> getCredentialsUnderPositionSync(int position) {
        return dao.getUnderPositionSync(position);
    }

    public void deleteCredentialsById(int id) {
        dao.deleteById(id);
    }

    public void nuke() {
        new Thread(database::clearAllTables).start();
    }
}

File: app/src/main/java/com/mirash/familiar/db/CredentialsRepository.kt
package com.mirash.familiar.db

import androidx.lifecycle.LiveData
import com.mirash.familiar.user.UserControl

/**
 * @author Mirash
 */
class CredentialsRepository(database: FamiliarDatabase) : IRepository<Credentials> {
    private val dao: CredentialsDao = database.getCredentialsDao()

    override fun insert(entity: Credentials): Long = dao.insert(entity)

    override fun update(entity: Credentials) {
        dao.update(entity)
    }

    override fun delete(entity: Credentials) {
        dao.delete(entity)
    }

    override fun deleteById(id: Long) {
        dao.deleteById(id)
    }

    override fun getById(id: Long): LiveData<Credentials> = dao.getById(id)

    override fun getByIdSync(id: Long): Credentials? = dao.getByIdSync(id)

    fun insertAll(credentialsList: List<Credentials>) {
        dao.insertAll(credentialsList)
    }

    fun updateAll(credentialsList: List<Credentials>) {
        dao.updateAll(credentialsList)
    }


    fun getAll(): LiveData<List<Credentials>> = dao.getAll()

    fun getAllSync(): List<Credentials> = dao.getAllSync()

    fun getAllByUserId(userId: Long): LiveData<List<Credentials>> =
        dao.getAllByUserId(userId)

    fun getCredentialsUnderPositionByUserIdSync(
        position: Int,
        userId: Long = UserControl.userId
    ): List<Credentials> =
        dao.getCredentialsUnderPositionByUserIdSync(position, userId)
}
